app/models/early_career_payments/slug_sequence.rb

```diff
@@ -99,6 +101,8 @@ def slugs
         sequence.delete("personal-bank-account") if claim.bank_or_building_society == "building_society"
         sequence.delete("building-society-account") if claim.bank_or_building_society == "personal_bank_account"
 
+        sequence.delete("teacher-reference-number") if session[:teacher_id]
```

Comment:
Okay, I will add the check for Teacher reference number.

For the flag on `Claim` I had sync with Martin, He said we are going to store data in session instead of database. Let me sync with him again and discuss this scenario.